force-app/main/default/classes/tasks/CollectionsTasks.cls

```diff
@@ -18,15 +24,31 @@ public class CollectionsTasks {
     Integer sum,
     List<Integer> listWithElementsToCheck
   ) {
-    return false;
+    boolean result = false;
+    for (integer i = 0; i < listWithElementsToCheck.size(); i++) {
+      for (integer j = i + 1; j < listWithElementsToCheck.size(); j++)
+        if (listWithElementsToCheck[i] + listWithElementsToCheck[j] == sum) {
+          result = true;
+          break;
+        }
+    }
+    return result;
   }
 
   // use single loop and an arbitrary collection to address same requirements as in sumCanBeReached
   public static Boolean sumCanBeReachedOptimal(
     Integer sum,
     List<Integer> listWithElementsToCheck
   ) {
-    return false;
+    boolean result = false;
```

Comment:
The idea is perfect! But there's a flaw in the algo. Try running this in apex anonymous:
 
```
Boolean result = CollectionsTasks.sumCanBeReachedOptimal(0, new List<Integer>{0, 1});
System.debug('Should be false. Actual: ' + result);
```

It should print out `Should be false. Actual: false`, but it returns true for your current implementation

Reply:
good catch! it returned null. Added line 59, see last commit